terraform-docs/terraform-docs
# Read config from relative path, absolute, or $HOME

### Description of your changes



Configuration can be loaded with `-c, --config string` which accepts
both relative and absolute paths.

    $ pwd
    /path/to/parent/folder
    $ tree
    .
    ├── module-a
    │   └── main.tf
    ├── module-b
    │   └── main.tf
    ├── ...
    └── .terraform-docs.yml

    # executing from parent
    $ terraform-docs -c .terraform-docs.yml module-a/

    # executing from child
    $ cd module-a/
    $ terraform-docs -c ../.terraform-docs.yml .

    # or an absolute path
    $ terraform-docs -c /path/to/parent/folder/.terraform-docs.yml .

The order for looking for config file is:

1. root of module directory
2. current directory
3. `$HOME/.tfdocs.d/`

if `.terraform-docs.yml` is found in any of the folders above, that will
take precedence and will override the other ones. Note that values passed
directly as CLI flags will override all of the above.

Fixes #445

I have:

- [x] Read and followed terraform-docs' [contribution process].
- [x] All tests pass when I run `make test`.

### How has this code been tested



Manually built and tested for various scenarios:

- config exists with default name
- config exists without default name and explicitly set with `--config`
- set config with `--config` with relative path (e.g. `../path/to/.terraform-docs.yml`)
- set config with `--config` with absolute path (e.g. `/absolute/path/to/.terraform-docs.yml`)
- create config file in 1) module directory and 2) its parent and 3) `$HOME/.tfdocs.d/` and make sure they will override one another

[contribution process]: https://git.io/JtEzg
